src/main/java/dal/fatih/todoly/repo/impl/JpaTaskRepositoryImpl.java

```diff
@@ -0,0 +1,101 @@
+package dal.fatih.todoly.repo.impl;
+
+import dal.fatih.todoly.model.Task;
+import dal.fatih.todoly.repo.TaskRepository;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import javax.persistence.metamodel.EntityType;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Locale;
+
+@Repository
+public class JpaTaskRepositoryImpl implements TaskRepository {

```

Comment:
No need to keep as Jpa anymore since it is the only option
```suggestion
public class TaskRepositoryImpl implements TaskRepository {

```